source/uri.c

```diff
@@ -327,18 +327,39 @@ static void s_parse_authority(struct uri_parser *parser, struct aws_byte_cursor
     struct aws_byte_cursor authority_parse_csr = parser->uri->authority;
 
     if (authority_parse_csr.len) {
+        /* RFC-3986 section 3.2: authority = [ userinfo "@" ] host [ ":" port ] */
+        uint8_t *userinfo_delim = memchr(authority_parse_csr.ptr, '@', authority_parse_csr.len);
+        if (userinfo_delim) {
+
+            parser->uri->userinfo =
+                aws_byte_cursor_advance(&authority_parse_csr, userinfo_delim - authority_parse_csr.ptr);
+            /* For the "@" mark */
+            aws_byte_cursor_advance(&authority_parse_csr, 1);
+            struct aws_byte_cursor userinfo_parse_csr = parser->uri->userinfo;
+            uint8_t *info_delim = memchr(userinfo_parse_csr.ptr, ':', userinfo_parse_csr.len);
+            /* RFC-3986 section 3.2.1: Use of the format "user:password" in the userinfo field is deprecated. But still,
+             * we will use this format here as it's a common use case. You also have the whole userinfo field. */
```

Comment:
TBD, I don't know what's the better way... Or we get a list of cursor for `userinfo = *( unreserved / pct-encoded / sub-delims / ":" )`? I feel like this way is better, but open to discussion.😃

Reply:
https://datatracker.ietf.org/doc/html/rfc1738#section-3.1 hmmm